Modules/System/Time Zone Management/src/DateTimeOffsetImpl.Codeunit.al

```diff
@@ -0,0 +1,37 @@
+codeunit 8721 "DateTime Offset Impl."
+{
+    Access = Internal;
+
+    procedure GetUtcOffset(SourceDateTime: DateTime; TimeZoneId: Text): Duration
+    var
+        TimeZoneInfoInitializer: Codeunit "TimeZoneInfo Initializer";
+        Offset: Duration;
+        TimeZoneInfoDotNet: DotNet TimeZoneInfo;
+    begin
+        TimeZoneInfoInitializer.InitializeTimeZoneInfo(TimeZoneId, TimeZoneInfoDotNet);
+        Offset := TimeZoneInfoDotNet.GetUtcOffset(SourceDateTime);
+        exit(Offset);
+    end;
+
+    procedure GetUtcOffsetForUserTimeZone(SourceDateTime: DateTime): Duration
+    var
+        Session: SessionSettings;
+    begin
+        Session.Init();
+        if Session.TimeZone <> '' then
```

Comment:
Can the Session have empty in TimeZone? If yes what happens in this case - Duration will be 0? If no is this line really needed?

Reply:
I believe it can be empty when called from a web service, for example. See procedures GetTimezoneOffset and GetUserClientTypeOffset for similar patterns.

That said, if I remove this statement, the missing time zone will be handled with an error in the initializer codeunit. That is a choice that we will have to make. Given the code in codeunit 10, I opted for similar patterning as in that codeunit.

Reply:
I have added the permission set to "System Application - Objects"; this seems the most logical place.